draft-ietf-lamps-dilithium-certificates.md

```diff
@@ -385,6 +385,59 @@ in this section.
 {{examples}} contains example ML-DSA private keys encoded using the
 textual encoding defined in {{RFC7468}}.
 
+# Pre-hashed mode (ExtarnalMu-ML-DSA)
+
+
+Many applications will require a "pre-hashed" mode of ML-DSA whereby a message digest can be pre-computed outside of the cryptographic module and then only a small fixed-width hash value is passed to the crypto module.
+Many applications and protocols include message digesting, but there exist some that do not. Examples of this can be found even within [RFC5280]; for example certificate and certificate revocation list (CRL) data structures do not include message digesting and therefore become problematic when producing large CRLs or when signing a high volume of certificates containing large public keys. Such situations require pre-hashing to be performed by the signature primitive.
+
+
+This section presents the "ExternalMu-ML-DSA" signature mode which provides an interface for performing pre-hashed signatures in a way which is both compliant with [FIPS204] and which produces signatures values which are indistinguishable from signatures produced by ML-DSA.Sign() and are therefore compatible with the normal ML-DSA.Verify() and are identified by the same Object Identifiers as for ML-DSA. A ML-DSA key and certificate MAY be used with either ML-DSA or ExternalMu-ML-DSA interchangeably. Note that ExternalMu-ML-DSA describes a different signature API from ML-DSA and therefore might require explicit support from hardware or software cryptographic modules.
+
+Note that the signing mode defined here is different from HashML-DSA defined in [FIPS204] section 5.4. This specification uses exclusively ExternalMu-ML-DSA for pre-hashed use cases, and thus HashML-DSA as defined in [FIPS204] and identified by `id-hash-ml-dsa-44-with-sha512`, `id-hash-ml-dsa-65-with-sha512`, and `id-hash-ml-dsa-87-with-sha512` MUST NOT be be used in X.509 and related PKIX protocols.
+
+The procedures below present modified versions of Algorithm 2 and Algorithm 7 in [FIPS204]. Note that all functions and notation used in {{fig-externalmu-ml-dsa-external}} and {{fig-externalmu-ml-dsa-internal}} are defined in [FIPS204].
```

Comment:
More specific references than 'Algorithm 2' and 'Algorithm 7'? e.g.

[Algorithm 2](https://nvlpubs.nist.gov/nistpubs/fips/nist.fips.204.pdf#algorithm.2) ML-DSA.Sign(𝑠𝑘, 𝑀, 𝑐𝑡𝑥) and [Algorithm7](https://nvlpubs.nist.gov/nistpubs/fips/nist.fips.204.pdf#algorithm.7) ML-DSA.Sign_internal(𝑠𝑘, 𝑀 ′, 𝑟𝑛𝑑)  in [FIPS204] 

(or similar)